### PR TITLE
Update README with outline, training, how to use the code

### PR DESCRIPTION
**Pull request by brikis98**
One of our customers asked how to use our code (e.g., how to consume a Terraform module) and the trade-offs between using the code directly from our repos vs forking. I've updated this ToC README accordingly. I also added an outline at the top and filled out the training materials section more.

### CODE REVIEW

**Comment by brikis98** [comment]
Merging now so I can send to customer. Feedback very welcome.

File: README.md [hunk 1]
```diff
@@ -244,8 +254,25 @@ In a multi-account setup, each environment (e.g., stage, prod, etc) is deployed
...
+
+1. [A Comprehensive Guide to Building a Scalable Web App on Amazon Web Services](https://www.airpair.com/aws/posts/building-a-scalable-web-app-on-amazon-web-services-p1): A definitive guide on how to think about building apps on AWS, including how to think about scalability and high availability, an overview of how to use the most important AWS services, and an introduction to cloud-native architecture.
+
+1. [A Comprehensive Guide to Authenticating to AWS on the Command Line](https://blog.gruntwork.io/a-comprehensive-guide-to-authenticating-to-aws-on-the-command-line-63656a686799): Our blog post series on how to authenticate to AWS on the command-line, including how to use Access Keys, IAM Roles, MFA, the Credentials File, Environment Eariables, Instance Metadata, and Gruntwork Houston.
```
Comment (yorinasub17) [inline README.md]: `Environment Eariables` => `Environment Variables`

File: README.md [hunk 2]
```diff
@@ -280,3 +307,72 @@ Here's a summary of the operating systems supported by Gruntwork as of January,
...
+
+1. **SSH for authentication**: If you [associate an SSH key with your GitHub account](https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account) and use [ssh-agent](https://help.github.com/en/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent), you will not be prompted for any passwords. You can use a similar approach with your machine users in CI / CD environments.
+
+**For binaries and scripts**, you can use the [gruntwork-installer](https://github.com/gruntwork-io/gruntwork-installer) with the `--repo` parameter pointing to Gruntwork repos. For example, to install the [ssh-grunt binary](https://github.com/gruntwork-io/module-security/tree/master/modules/ssh-grunt) (which lets you manage SSH access using an Identity Provider such as IAM, Google, ADFS, etc) on your EC2 Instances:
```
Comment (yorinasub17) [inline README.md]: Might want to mention that `gruntwork-installer` is a bash script, and thus won't work with Windows.

**Comment by yorinasub17** [comment]
LGTM!

**Comment by josh-padnick** [comment]
Lots of great content here. Thanks for writing! 👍 

But...will users know to even come here? I worry that we are once again seeing the need for an authoritative "starting point" for our new customers.  Let's discuss offline.

File: README.md (hunk 1 above)
Comment (brikis98) [reply README.md]: Fixed, thanks! https://github.com/gruntwork-io/toc/commit/0723f2437cf1c4341c73ee91212713ba5096644a

File: README.md (hunk 2 above)
Comment (brikis98) [reply README.md]: Fixed, thanks! https://github.com/gruntwork-io/toc/commit/0723f2437cf1c4341c73ee91212713ba5096644a

**Comment by brikis98** [comment]
> But...will users know to even come here? I worry that we are once again seeing the need for an authoritative "starting point" for our new customers. Let's discuss offline.

Yea, I wasn't sure where to put this. We do send the ToC link in the "welcome" email to all users, so this is sort of a starting point, but very far from ideal. Definitely worth discussing further.

**Pull request merged**